tests/appsec/test_asm_standalone.py

```diff
@@ -863,6 +861,155 @@ def propagated_tag_value(self):
+        assert self.request1.status_code == 200
+        self.verify_trace_sampling(self.request1)
+
+        assert self.request2.status_code == 200
+        self.verify_trace_sampling(self.request2)
+
+        # Subsequent requests to same endpoint sampled out
+        for request in self.subsequent_requests:
+            assert request.status_code == 200
+            self.verify_trace_sampling(request, sampling_priority=-1, should_have_schema=False)
+
+    def setup_sampling_window_renewal(self):
+        time.sleep(4)  # Wait for the sampling window to expire
+
+        self.endpoint = "/api_security/sampling/200"
+        self.window1_request1 = weblog.get(self.endpoint, headers=self._get_headers())
+        self.window1_request2 = weblog.get(self.endpoint, headers=self._get_headers())
+        time.sleep(4)  # Delay is set to 3s via the env var DD_API_SECURITY_SAMPLE_DELAY
+        self.window2_request1 = weblog.get(self.endpoint, headers=self._get_headers())
+
+    def test_sampling_window_renewal(self):
+        """Verify that endpoint sampling resets after the sampling window expires"""
+
+        # First request should be retained with schema
+        assert self.window1_request1.status_code == 200
+        self.verify_trace_sampling(self.window1_request1)
+
+        # Following request sampled out
+        assert self.window1_request2.status_code == 200
+        self.verify_trace_sampling(self.window1_request2, sampling_priority=-1, should_have_schema=False)
+
+        # After window expiration, request retained again
+        assert self.window2_request1.status_code == 200
+        self.verify_trace_sampling(self.window2_request1)
+
+    def setup_user_keep_priority_does_not_force_schema_collection(self):
+        """Test that spans with USER_KEEP priority do not force schema collection"""
+
+        time.sleep(4)  # Wait for the sampling window to expire
+
+        self.endpoint = "/api_security/sampling/200"
+
+        # Set USER_KEEP (2) priority explicitly
+        headers = self._get_headers()
+        headers["x-datadog-sampling-priority"] = "2"  # USER_KEEP
+
+        # Make multiple requests to same endpoint that would normally be sampled out
+        self.request1 = weblog.get(self.endpoint, headers=headers)
+        self.request2 = weblog.get(self.endpoint, headers=headers)
+        self.request3 = weblog.get(self.endpoint, headers=headers)
+
+    def test_user_keep_priority_does_not_force_schema_collection(self):
+        """Test that spans with USER_KEEP priority do not force schema collection"""
+
+        assert self.request1.status_code == 200
+        self.verify_trace_sampling(self.request1, sampling_priority=2)
+
+        assert self.request2.status_code == 200
+        self.verify_trace_sampling(self.request2, sampling_priority=-1, should_have_schema=False)
+
+        assert self.request3.status_code == 200
+        self.verify_trace_sampling(self.request3, sampling_priority=-1, should_have_schema=False)
```

Comment:
we should probably be expecting `sampling_priority=2, should_have_schema=False` here